File: bindings/python/examples/how_tos/alias_wallet/transaction.py
from iota_sdk import Wallet, Utils, NodeIndexerAPI, SyncOptions, AliasSyncOptions, SendParams
from dotenv import load_dotenv
import os

# In this example we send funds from an alias wallet.

load_dotenv()

sync_options = SyncOptions(alias=AliasSyncOptions(basic_outputs=True))

wallet = Wallet(os.environ['WALLET_DB_PATH'])

account = wallet.get_account('Alice')

if 'STRONGHOLD_PASSWORD' not in os.environ:
    raise Exception(".env STRONGHOLD_PASSWORD is undefined, see .env.example")

wallet.set_stronghold_password(os.environ["STRONGHOLD_PASSWORD"])

balance = account.sync(sync_options)

total_base_token_balance = balance.baseCoin.total
print(f'Balance before sending funds from alias: {total_base_token_balance}')

alias_id = balance.aliases[0]
print(f'Alias Id: {alias_id}')

# Get alias address
alias_address = Utils.alias_id_to_bech32(
    alias_id, wallet.get_client().get_bech32_hrp())

# Find first output unlockable by the alias address
query_parameters = NodeIndexerAPI.QueryParameters(alias_address)
input = wallet.get_client().basic_output_ids(query_parameters).items[0]

params = [SendParams(
    address='rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu',
    amount=1000000,
)]
options = {
    'mandatoryInputs': [input],
}
transaction = account.send(params, options)
account.retry_transaction_until_included(transaction.transactionId, None, None)
print(
    f'Transaction with custom input: https://explorer.shimmer.network/testnet/transaction/{transaction.transactionId}')

total_base_token_balance = account.sync(sync_options).baseCoin.total
print(f'Balance after sending funds from alias: {total_base_token_balance}')

File: bindings/nodejs/examples/how_tos/alias_wallet/transaction.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Wallet, initLogger, Utils } from '@iota/sdk';

// This example uses secrets in environment variables for simplicity which should not be done in production.
//
// Make sure that `example.stronghold` and `example.walletdb` already exist by
// running the `how_tos/accounts_and_addresses/create-account` example!
//
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./how_tos/alias_wallet/transaction.ts

// In this example we send funds from an alias wallet.
async function run() {
    initLogger();
    try {
        for (const envVar of ['WALLET_DB_PATH', 'STRONGHOLD_PASSWORD']) {
            if (!(envVar in process.env)) {
                throw new Error(
                    `.env ${envVar} is undefined, see .env.example`,
                );
            }
        }

        const syncOptions = {
            alias: {
                basicOutputs: true,
            },
        };

        const wallet = new Wallet({
            storagePath: process.env.WALLET_DB_PATH,
        });

        const account = await wallet.getAccount('Alice');

        await wallet.setStrongholdPassword(
            process.env.STRONGHOLD_PASSWORD as string,
        );

        const balance = await account.sync(syncOptions);

        const totalBaseTokenBalance = balance.baseCoin.total;
        console.log(
            `Balance before sending funds from alias: ${totalBaseTokenBalance}`,
        );

        const aliasId = balance.aliases[0];
        console.log(`Alias Id: ${aliasId}`);

        // Get Alias address
        const aliasAddress = Utils.aliasIdToBech32(
            aliasId,
            await (await wallet.getClient()).getBech32Hrp(),
        );

        // Find first output unlockable by the alias address
        const queryParameters = [
            {
                address: aliasAddress,
            },
        ];
        const input = (
            await (await wallet.getClient()).basicOutputIds(queryParameters)
        ).items[0];

        const params = [
            {
                address:
                    'rms1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluaw60xu',
                amount: BigInt(1000000),
            },
        ];
        const options = {
            mandatoryInputs: [input],
            allowMicroAmount: false,
        };
        const transaction = await account.sendWithParams(params, options);
        await account.retryTransactionUntilIncluded(transaction.transactionId);
        console.log(
            `Transaction with custom input: https://explorer.iota.org/testnet/transaction/${transaction.transactionId}`,
        );

        const totalBaseTokenBalanceAfter = (await account.sync(syncOptions))
            .baseCoin.total;
        console.log(
            `Balance after sending funds from alias: ${totalBaseTokenBalanceAfter}`,
        );
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
